fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java

```diff
@@ -548,6 +763,7 @@ public Expression toExpression() {
                 return toExpr;
             }
             Expression result = sourceValues.get(0).toExpression();
+            BinaryOperator<Expression> mergeExprOp = isAnd ? ExpressionUtils::and : ExpressionUtils::or;
             for (int i = 1; i < sourceValues.size(); i++) {
                 result = mergeExprOp.apply(result, sourceValues.get(i).toExpression());
```

Comment:
avoid to generate deep tree
            list = new List(result);
            for (int i = 1; i < sourceValues.size(); i++) {
                list.add(sourceValues.get(i).toExpression())
            }
            result = new And(list);

Reply:
fix